mm2src/trezor/src/transport/udp.rs

```diff
@@ -0,0 +1,161 @@
+/// This source file was borrowed from Trezor repo (https://raw.githubusercontent.com/trezor/trezor-firmware/07ba960ab4aa5aa3ddf16ae74c3658782d491250/rust/trezor-client/src/transport/udp.rs)
+/// and modified to integrate into this project.
+/// Adds udp transport to interact with trezor emulator.
+/// To build emulator use this repo: https://github.com/trezor/trezor-firmware, build with build-docker.sh for the desired branch (tag)
+/// Tested with the legacy emulator (for Trezor One).
+/// After building the emulator find it as ./legacy/firmware/trezor.elf file.
+/// Start it (no params needed) and initialize it with Trezor Suite: create a wallet, find a receive address.
+/// You need the bridge for connecting from the Suite, it can be downloaded from trezor.io.
+/// Do not use pin for the created wallet.
+/// Be aware that when you rebuild the firmware the emulator flash memory file emulator.img is recreated (so save it before rebuilding code)
+use super::{protocol::{Link, Protocol, ProtocolV1},
+            ProtoMessage, Transport};
+use crate::{TrezorError, TrezorResult};
+use async_std::{io, net::UdpSocket, task::block_on};
+use mm2_err_handle::prelude::*;
+use std::time::Duration;
+
+// A collection of constants related to the Emulator Ports.
+mod constants {
+    pub const DEFAULT_HOST: &str = "127.0.0.1";
+    pub const DEFAULT_PORT: &str = "21324";
+    pub const DEFAULT_DEBUG_PORT: &str = "21325";
+    pub const LOCAL_LISTENER: &str = "127.0.0.1:0";
```

Comment:
`pub(super)` for visibility in the parent module only, or you can move them to be with the other constants in the parent module and make them private to it.

Reply:
added super (I guess I like to have them where they originally were added by the ported code)